### ISSUE TITLE
nat_loss.py label smoothing

### ISSUE THREAD

**Issue by haorannlp** [facebookresearch/fairseq#3911]
## 🐛 Bug

In line 68 of `nat_loss.py`, the label smoothed nat loss is computed as follows:

```
loss = nll_loss * (1 - label_smoothing) - mean_ds(logits) * label_smoothing
                
```
I think the smoothing part should be `mean_ds(logits) * label_smoothing / (logits.size(-1)-1)`

### To Reproduce

Steps to reproduce the behavior (**always include the command you ran**):

1. Run cmd '....'
2. See error

### Environment

 - fairseq Version (e.g., 1.0 or main):
 - PyTorch Version (e.g., 1.0)
 - OS (e.g., Linux):
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by haorannlp** [comment]
mean_ds computes mean of all the elements, so it is correct
